diff --git a/absl/base/CMakeLists.txt b/absl/base/CMakeLists.txt
index 84decab2..66162aa2 100644
--- a/absl/base/CMakeLists.txt
+++ b/absl/base/CMakeLists.txt
@@ -266,7 +266,7 @@ absl_cc_library(
     "casts.cc"
     "internal/cycleclock.cc"
     "internal/spinlock.cc"
-    "internal/sysinfo.cc"
+#    "internal/sysinfo.cc"
     "internal/thread_identity.cc"
     "internal/unscaledcycleclock.cc"
   COPTS
diff --git a/absl/debugging/CMakeLists.txt b/absl/debugging/CMakeLists.txt
index ab3a7953..f481c76b 100644
--- a/absl/debugging/CMakeLists.txt
+++ b/absl/debugging/CMakeLists.txt
@@ -174,7 +174,7 @@ absl_cc_library(
   HDRS
     "failure_signal_handler.h"
   SRCS
-    "failure_signal_handler.cc"
+#    "failure_signal_handler.cc"
   COPTS
     ${ABSL_DEFAULT_COPTS}
   DEPS
@@ -216,8 +216,8 @@ absl_cc_library(
     "internal/vdso_support.h"
   SRCS
     "internal/address_is_readable.cc"
-    "internal/elf_mem_image.cc"
-    "internal/vdso_support.cc"
+#    "internal/elf_mem_image.cc"
+#    "internal/vdso_support.cc"
   COPTS
     ${ABSL_DEFAULT_COPTS}
   DEPS
